src/sentry/preprod/api/endpoints/organization_preprod_app_size_stats.py

```diff
@@ -0,0 +1,407 @@
+            "start": unix_timestamp,
+            "end": unix_timestamp,
+            "meta": {...},
+            "filters": {  // Only if includeFilters=true
+                "app_ids": ["com.example.app", ...],
+                "branches": ["main", "develop", ...]
+            }
+        }
+        """
+        try:
+            project_ids_set = self.get_requested_project_ids_unchecked(request)
+            if not project_ids_set:
+                projects = self.get_projects(request, organization)
+                project_ids = [p.id for p in projects]
+            else:
+                project_ids = list(project_ids_set)
+
+            # Use shared utility for time range parsing
+            start, end = get_date_range_from_params(
+                request.GET, default_stats_period=timedelta(days=14)
+            )
+
+            # Use shared utility for interval parsing
+            interval_seconds = get_rollup_from_request(
+                request,
+                date_range=end - start,
+                default_interval="1d",
+                error=ParseError("Invalid interval"),
+            )
+
+            field = request.GET.get("field", "max(max_install_size)")
+            aggregate_func, aggregate_field = self._parse_field(field)
+
+            filter_kwargs = self._parse_query(request.GET.get("query", ""))
+            query_filter = self._build_filter(filter_kwargs) if filter_kwargs else None
+
+            response = query_preprod_size_metrics(
+                organization_id=organization.id,
+                project_ids=project_ids,
+                start=start,
+                end=end,
+                referrer="api.preprod.app-size-stats",
+                filter=query_filter,
+                limit=10000,
+            )
+
+            timeseries_data = self._transform_to_timeseries(
+                response, start, end, interval_seconds, aggregate_func, aggregate_field
+            )
+
+            result: dict[str, Any] = {
+                "data": timeseries_data,
+                "start": int(start.timestamp()),
+                "end": int(end.timestamp()),
+                "meta": {
+                    "fields": {
+                        field: "integer",
+                    }
+                },
+            }
+
+            include_filters = request.GET.get("includeFilters", "").lower() == "true"
+            if include_filters:
+                filter_values = self._fetch_filter_values(organization.id, project_ids, start, end)
+                result["filters"] = filter_values
+
+            return Response(result)
+
+        except InvalidParams:
+            logger.exception("Invalid parameters for app size stats request")
+            raise ParseError("Invalid query parameters")
+        except (ValueError, KeyError):
+            logger.exception("Error while parsing app size stats request")
+            raise ParseError("Invalid request parameters")
```

Comment:
Is it possible to shrink the code which lives inside the try catch block to specifically what could potentially error?